graphs/strongly_connected_components.py

```diff
@@ -0,0 +1,117 @@
+"""
+https://en.wikipedia.org/wiki/Strongly_connected_component
+
+Finding strongly connected components in directed graph
+
+"""
+
+test_graph_1 = {
+    0: [2, 3],
+    1: [0],
+    2: [1],
+    3: [4],
+    4: [],
+}
+
+test_graph_2 = {
+    0: [1, 2, 3],
+    1: [2],
+    2: [0],
+    3: [4],
+    4: [5],
+    5: [3],
+}
+
+
+def topology_sort(graph: dict, vert: int, visited: list) -> list:
+    """
+    Use depth first search to sort graph
+    At this time graph is the same as input
+    >>> topology_sort(test_graph_1, 0, 5 * [False])
+    [1, 2, 4, 3, 0]
+    >>> topology_sort(test_graph_2, 0, 6 * [False])
+    [2, 1, 5, 4, 3, 0]
+    """
+
+    visited[vert] = True
+    order = []
+
+    for neighbour in graph[vert]:
+        if not visited[neighbour]:
+            order += topology_sort(graph, neighbour, visited)
+
+    order.append(vert)
+
+    return order
+
+
+def find_components(reverse_graph: dict, vert: int, visited: list) -> list:
+    """
+    Use depth first search to find strongliy connected
+    vertices. Now graph is reversed
+    >>> find_components({0: [1], 1: [2], 2: [0]}, 0, 5 * [False])
+    [0, 1, 2]
+    >>> find_components({0: [2], 1: [0], 2: [0, 1]}, 0, 6 * [False])
+    [0, 2, 1]
+    """
+
+    visited[vert] = True
+
+    component = [vert]
+
```

Comment:
`visited[vert], component = True, [vert]`

Reply:
I find the current code more readable than the proposed code but we could remove the blank line between the two statements.